dusty/commands/run.py

```diff
@@ -85,3 +86,13 @@ def restart_apps_or_services(app_or_service_names=None, sync=True):
             rsync.sync_repos(spec_assembler.get_all_repos(active_only=True, include_specs_repo=False))
 
     compose.restart_running_services(app_or_service_names)
+
+def restart_apps_by_repo(repo_names, sync=True):
+    all_active_repos = spec_assembler.get_all_repos()
+    resolved_repos = set([Repo.resolve(all_active_repos, repo_name) for repo_name in repo_names])
+    specs = spec_assembler.get_assembled_specs()
+    apps_with_repos = set()
+    for app_spec in specs['apps'].itervalues():
+        if spec_assembler.get_same_container_repos_from_spec(app_spec).intersection(resolved_repos):
```

Comment:
This is likely to break at some point because it explicitly depends on the return type being a set. I think most Python programmers are used to sequences being used interchangeably, it wouldn't surprise me if someone changed it to a list at some point.

This should be tested, can you write integration tests for it?